apiserver/facades/agent/machineactions/shim.go

```diff
@@ -23,7 +23,12 @@ type backendShim struct {
 }
 
 func (shim backendShim) ActionByTag(tag names.ActionTag) (state.Action, error) {
-	return shim.st.ActionByTag(tag)
+	m, err := shim.st.Model()
+	if err != nil {
+		return nil, err
+	}
+
+	return m.ActionByTag(tag)
```

Comment:
I imagine that soon the `Backend` interface here will be based around `state.Model` instead of `state.State` but this is fine for now.